### PR TITLE
[IR-3426] Unit Test - Spatial Transform coverage to 80%

### PR DESCRIPTION
**Pull request by heysokam**
## References
https://tsu.atlassian.net/browse/IR-3426

### CODE REVIEW

File: packages/spatial/src/transform/components/TransformComponent.test.tsx [hunk 1]
```diff
@@ -23,27 +23,1188 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
...
+      matrixWorld[12].set(Position.x)
+      matrixWorld[13].set(Position.y)
+      matrixWorld[14].set(Position.z)
+      // Run and Check the result
+      const result = new Matrix4()
+      TransformComponent.getMatrixRelativeToScene(testEntity, result)
+      assertArrayEqual(result.elements, Expected.elements)
+    })
+
+    it('should return `@param entity`.TransformComponent.matrixWorld when neither itself nor one of its ancestors have a SceneComponent', () => {
```
Comment (HexaField) [inline packages/spatial/src/transform/components/TransformComponent.test.tsx:343]: duplicate test as above

**Review by MichaelEstes — APPROVED** [review]

File: packages/spatial/src/physics/systems/PhysicsSystem.test.ts [hunk 2]
```diff
@@ -298,6 +297,7 @@ describe('PhysicsSystem', () => {
 
       const physicsSystemReactor = SystemDefinitions.get(PhysicsSystem)?.reactor
 
+      /** @todo Why is the world not recreated as expected ?? */
```
Comment (HexaField) [inline packages/spatial/src/physics/systems/PhysicsSystem.test.ts:300]: i tested this myself and it seems to be fine. what error are you seeing?

File: packages/spatial/src/transform/systems/LookAtSystem.test.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        const before = getComponent(testEntity, TransformComponent).rotation.clone()
+        assertVecApproxEq(before, Initial, 4)
+        // Run and Check the result
+        System.execute()
+        const result = getComponent(testEntity, TransformComponent).rotation.clone()
+        assertVecApproxEq(result, Initial, 4)
+      })
+    })
+
+    /** @todo */
```
Comment (HexaField) [inline packages/spatial/src/transform/systems/LookAtSystem.test.ts]: this looks done, forgot to remove todo?

**Review by HexaField — APPROVED** [review]
Looks amazing. Great work.

File: packages/spatial/src/physics/systems/PhysicsSystem.test.ts (hunk 2 above)
Comment (heysokam) [reply packages/spatial/src/physics/systems/PhysicsSystem.test.ts:300]: It's a retroactive note. I saw the `.skip` but I never let skips in without a todo, so added it.

If I remove the `.skip` from the test right below this line, I get this error:
```ts
AssertionError: Got unwanted exception.
Actual message: "Physics world not found"
 ❯ src/physics/systems/PhysicsSystem.test.ts:310:16
    308|         root.run()
    309|         assert.equal(hasComponent(physicsWorldEntity, SceneComponent), true)
    310|         assert.doesNotThrow(() => Physics.destroyWorld(getComponent(physicsWorldEntity, UUIDComponent)))
       |                ^
    311|       })
    312|     }) //:: PhysicsSceneReactor
```

File: packages/spatial/src/transform/systems/LookAtSystem.test.ts (hunk 3 above)
Comment (heysokam) [reply packages/spatial/src/transform/systems/LookAtSystem.test.ts]: Fixed with b7db340
Good catch, ty!

**Pull request merged**